Wiki Categories Revamp & Tags Sorting

Yomzy2223 opened the pull request:
# Wiki Categories Revamp & Tags Sorting

_This PR updates the Wiki Categories UI and introduces a new feature that allows users to sort wikis based on tags. This enhancement will help users find content more efficiently and provide a better browsing experience._

## How should this be tested?

1. Navigate to wiki edit page, and click the "Edit wiki details" button
2. Select any of the categories, then add some tags, and publish
3. Navigate to the category, then trying filtering with the tags

## Notes or observations

1. _On the wiki details modal, tags list is now dependent on the selected category (i.e changes with category)_
2. _Added some new tags._
3. _The filter feature has currently been commented out until editors are done adding the new tags to the wikis. Can test the filter feature here https://iq-wiki-p5tlk12w7-prediqt.vercel.app/categories/nfts/

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/3040

Tolu-Mals commented:
I'm seeing this weird behaviour where I can select at most 4 tags, some other times at it becomes 5 tags. sometimes I can't click away from a particular tag. Then there was a time I couldn't select any tag except I clicked on 'All' first. I'll share the video soon

Tolu-Mals commented:
https://app.usebubbles.com/roa3LpdTPSRDt25QdBoFBV/can-t-select-at-most-4-tags

Tolu-Mals commented:
https://app.usebubbles.com/6fqsDP9ozXJWa6ZUKTPD3v/can-t-select-any-tag-till-select-all

Tolu-Mals commented:
https://app.usebubbles.com/hDiW4YD44QPrFPYm7wfLQ7/can-t-deselect-first-tag

Tolu-Mals commented:
https://app.usebubbles.com/5f1grqSHbn6GqsBfrhqRs3/can-t-remove-selected-tag

Yomzy2223 commented:
> I'm seeing this weird behaviour where I can select at most 4 tags, some other times at it becomes 5 tags. sometimes I can't click away from a particular tag. Then there was a time I couldn't select any tag except I clicked on 'All' first. I'll share the video soon

Hmm... weird! Kindly share the video, thanks.

Tolu-Mals commented:
> > I'm seeing this weird behaviour where I can select at most 4 tags, some other times at it becomes 5 tags. sometimes I can't click away from a particular tag. Then there was a time I couldn't select any tag except I clicked on 'All' first. I'll share the video soon
> 
> Hmm... weird! Kindly share the video, thanks.

yeah check the links

Yomzy2223 commented:
> https://app.usebubbles.com/6fqsDP9ozXJWa6ZUKTPD3v/can-t-select-any-tag-till-select-all

Looks like the page had not finished loading here

Aliiiu commented:
> 3\. https://iq-wiki-4zbovowel-prediqt.vercel.app/categories/nfts/

@Yomzy2223 If the max tags I can select is 4 just disabled the remaining tags or use the `cursor-not-allowed` class

Aliiiu commented:
> > > https://app.usebubbles.com/roa3LpdTPSRDt25QdBoFBV/can-t-select-at-most-4-tags
> > 
> > 
> > @Yomzy2223 If the max tags I can select is 4 just disable the remaining tags or use the `cursor-not-allowed` class
> 
> True true

<img width="1478" alt="image" src="https://github.com/user-attachments/assets/aca0a9fd-ef35-4e80-bf46-752c50920a23">
THis is nice but when I hover on it is not showing that the cursor is not allowed. Don't forget to use the class

Aliiiu commented on src/pages/categories/[category].tsx:
```diff
@@ -79,6 +120,34 @@ const CategoryPage = ({
   })
   const { t } = useTranslation('category')
+
+  const _handleTags = (updatedTags: string[]) => {
```
Don't forget to resolve the issue with this function

Yomzy2223 replied:
Was intentional. Was because the tags are currently commented out.

Aliiiu commented:
<img width="2053" alt="image" src="https://github.com/user-attachments/assets/5245b86a-c553-4f55-8cc5-12e90fa2545e">

How to recreate
- Go to create-wiki
- open the edit wiki modal

Yomzy2223 commented:
> > > > https://app.usebubbles.com/roa3LpdTPSRDt25QdBoFBV/can-t-select-at-most-4-tags
> > > 
> > > 
> > > @Yomzy2223 If the max tags I can select is 4 just disable the remaining tags or use the `cursor-not-allowed` class
> > 
> > 
> > True true
> 
> <img alt="image" width="1478" src="https://private-user-images.githubusercontent.com/67792966/363138408-aca0a9fd-ef35-4e80-bf46-752c50920a23.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMTQwMzgsIm5iZiI6MTcyNTAxMzczOCwicGF0aCI6Ii82Nzc5Mjk2Ni8zNjMxMzg0MDgtYWNhMGE5ZmQtZWYzNS00ZTgwLWJmNDYtNzUyYzUwOTIwYTIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDEwMjg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjZjg5YWM3MmM1ZDcxNjI2YmJiMDQ3MTg5OTA2YTlkNGNiMjE3OTEzNzU0YWEzMTVkNzI5YmFjYTk0NDQwYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.f1lG00Cp4AXlQFakeOgwp-tsmKQ2U9Y359_6zYWgbsE"> THis is nice but when I hover on it is not showing that the cursor is not allowed. Don't forget to use the class

Yeah. Now added. Also, I think it makes sense to not disable All button when you max out, so I fixed that too. Can take a look here https://iq-wiki-p5tlk12w7-prediqt.vercel.app/categories/nfts/

Aliiiu commented:
> https://iq-wiki-p5tlk12w7-prediqt.vercel.app/categories/nfts/

Nice, looks nice

Yomzy2223 commented:
> <img alt="image" width="2000" src="https://private-user-images.githubusercontent.com/67792966/363163566-5245b86a-c553-4f55-8cc5-12e90fa2545e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMjQzMTUsIm5iZiI6MTcyNTAyNDAxNSwicGF0aCI6Ii82Nzc5Mjk2Ni8zNjMxNjM1NjYtNTI0NWI4NmEtYzU1My00ZjU1LThjYzUtMTJlOTBmYTI1NDVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDEzMjAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiNWRkMWVmYmZiN2M1YzM3Y2YwNjQ3OGMxYjliYWU3MDA2YTg0YmNlZTQwOTNhODBkMzQyZTliNTgwNzgxNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RgoJUE7uGtQQ4BUfEQYqtrpRlLq1xRHiyvSSQzZQW5w">
> How to recreate
> 
> * Go to create-wiki
> * open the edit wiki modal

Fixed now

Aliiiu commented:
> > <img alt="image" width="2000" src="https://private-user-images.githubusercontent.com/67792966/363163566-5245b86a-c553-4f55-8cc5-12e90fa2545e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMjQzMTUsIm5iZiI6MTcyNTAyNDAxNSwicGF0aCI6Ii82Nzc5Mjk2Ni8zNjMxNjM1NjYtNTI0NWI4NmEtYzU1My00ZjU1LThjYzUtMTJlOTBmYTI1NDVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDEzMjAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiNWRkMWVmYmZiN2M1YzM3Y2YwNjQ3OGMxYjliYWU3MDA2YTg0YmNlZTQwOTNhODBkMzQyZTliNTgwNzgxNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RgoJUE7uGtQQ4BUfEQYqtrpRlLq1xRHiyvSSQzZQW5w">
> > How to recreate
> > 
> > * Go to create-wiki
> > * open the edit wiki modal
> 
> Fixed now

<img width="1480" alt="image" src="https://github.com/user-attachments/assets/a8563c89-3f4a-4d9e-be53-bb65085dc0b2">

Same with the dropdown for link wikis and links

Aliiiu commented:
<img width="661" alt="image" src="https://github.com/user-attachments/assets/e475a85d-652a-4e7b-adcc-fec0af4dc7ea">

@Simplypheyie96 is the dropdown suppose to be this close?

Simplypheyie96 commented:
> <img alt="image" width="661" src="https://private-user-images.githubusercontent.com/67792966/363245258-e475a85d-652a-4e7b-adcc-fec0af4dc7ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzM5NzEsIm5iZiI6MTcyNTAzMzY3MSwicGF0aCI6Ii82Nzc5Mjk2Ni8zNjMyNDUyNTgtZTQ3NWE4NWQtNjUyYS00ZTdiLWFkY2MtZmVjMGFmNGRjN2VhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDE2MDExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmOGVlNzQxZGNhN2YyODE0ZWVhZjg3Yzc4MTZiNDBkZmJjMDllYWY0MTcyMTZjNTU1MWM2NDI1MGI3ZGM3YzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LdmkFKt3EnoFEKVWHfaIYTRMf9YomewEHUsbAhbOmyY">
> @Simplypheyie96 is the dropdown supposed to be this close?

These are not issues that just arised on this pr, Except we want to fix them in this pr.

Aliiiu commented:
> > <img alt="image" width="661" src="https://private-user-images.githubusercontent.com/67792966/363245258-e475a85d-652a-4e7b-adcc-fec0af4dc7ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzM5NzEsIm5iZiI6MTcyNTAzMzY3MSwicGF0aCI6Ii82Nzc5Mjk2Ni8zNjMyNDUyNTgtZTQ3NWE4NWQtNjUyYS00ZTdiLWFkY2MtZmVjMGFmNGRjN2VhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDE2MDExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmOGVlNzQxZGNhN2YyODE0ZWVhZjg3Yzc4MTZiNDBkZmJjMDllYWY0MTcyMTZjNTU1MWM2NDI1MGI3ZGM3YzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LdmkFKt3EnoFEKVWHfaIYTRMf9YomewEHUsbAhbOmyY">
> > @Simplypheyie96 is the dropdown supposed to be this close?
> 
> These are not issues that just arised on this pr, Except we want to fix them in this pr.

True, i will look into it in another PR @Softdev1
